go-controller/pkg/ovn/controller/unidling/unidle.go

```diff
@@ -91,66 +125,81 @@ type ServiceVIPKey struct {
 	// Load balancer VIP in the form "ip:port"
 	vip string
 	// Protocol used by the load balancer
-	protocol v1.Protocol
+	protocol kapi.Protocol
 }
 
 // AddServiceVIPToName associates a k8s service name with a load balancer VIP
-func (uc *unidlingController) AddServiceVIPToName(vip string, protocol v1.Protocol, namespace, name string) {
+func (uc *unidlingController) AddServiceVIPToName(vip string, protocol kapi.Protocol, namespace, name string) {
 	uc.serviceVIPToNameLock.Lock()
 	defer uc.serviceVIPToNameLock.Unlock()
 	uc.serviceVIPToName[ServiceVIPKey{vip, protocol}] = types.NamespacedName{Namespace: namespace, Name: name}
 }
 
 // GetServiceVIPToName retrieves the associated k8s service name for a load balancer VIP
-func (uc *unidlingController) GetServiceVIPToName(vip string, protocol v1.Protocol) (types.NamespacedName, bool) {
+func (uc *unidlingController) GetServiceVIPToName(vip string, protocol kapi.Protocol) (types.NamespacedName, bool) {
 	uc.serviceVIPToNameLock.Lock()
 	defer uc.serviceVIPToNameLock.Unlock()
 	namespace, ok := uc.serviceVIPToName[ServiceVIPKey{vip, protocol}]
 	return namespace, ok
 }
 
 // DeleteServiceVIPToName retrieves the associated k8s service name for a load balancer VIP
-func (uc *unidlingController) DeleteServiceVIPToName(vip string, protocol v1.Protocol) {
+func (uc *unidlingController) DeleteServiceVIPToName(vip string, protocol kapi.Protocol) {
 	uc.serviceVIPToNameLock.Lock()
 	defer uc.serviceVIPToNameLock.Unlock()
 	delete(uc.serviceVIPToName, ServiceVIPKey{vip, protocol})
 }
 
 func (uc *unidlingController) Run(stopCh <-chan struct{}) {
-	ticker := time.NewTicker(5 * time.Second)
-
+	ctx := context.Background()
```

Comment:
can you use a context with a the regular ovsdb timeout like we've updated other parts of the code to use?